icecreamshop/icecreamshop/src/main/java/com/summerschool/icecreamshop/controller/ProductController.java

```diff
@@ -23,4 +21,23 @@ public ResponseEntity<Product> readById(@PathVariable Integer productId) {
         Optional<Product> optionalProduct = productService.findById(productId);
         return optionalProduct.map(product -> new ResponseEntity<>(product, HttpStatus.OK)).orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
     }
+
+    @PatchMapping("/{productId}")
+    public ResponseEntity<Product> update(@RequestBody Product product) {
+        if(product.getTitle() != null &&
```

Comment:
Please add @notblank or NotNull annotation over each object and remove the if
NotBlank - checks both null and empty and used for String type
NotNull - for other types to check if null.

https://www.baeldung.com/java-bean-validation-not-null-empty-blank